library/src/cache/patch_manager.rs

```diff
@@ -179,9 +190,52 @@ impl PatchManager {
             );
         }
 
+        // Verify that we haven't tried and failed to boot this patch.
+        if self.is_patch_last_attempted_patch(patch.number) {
+            match self.last_successful_boot_patch_number() {
```

Comment:
last_successful_boot_patch_number still feels like the wrong check.  We care if this patch is a known good patch or not (have we *ever* successfully booted from it), not if the last patch we happened to boot from was this one.

Reply:
This was my confusion thinking we still had a known_good and known_bad list.  In our current updater whenever we record a new "successful" patch we forget all knowledge about past patches good or bad state.